lib/queue.rb

```diff
@@ -1,31 +1,77 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(20)
+    @size = 20
+    @front = -1
+    @rear = -1
+  
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+
+    if (@front == 0 && @rear == @size - 1) ||
+      (@rear == (@front - 1) % (@size - 1))
+      raise ArgumentError.new("Error: Queue is full")
+    elsif @front == -1 # queue is empty
+      @front = @rear = 0
+      @store[@rear] = element
+    elsif @rear == @size - 1 && @front != 0 # rear needs to wrap around
+      @rear = 0
+    else 
+      @rear += 1
+    end
+    @store[@rear] = element
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1
+      raise RuntimeError.new("The queue is empty")
+    end
+
+    removed_data = @store[@front]
+    @store[@front] = nil
+
+    if @front == @rear # now queue is empty
+      @front = -1
+      @rear = -1
+    elsif @front == @size - 1
+      @front = 0
+    else 
+      @front += 1
+    end
+    return removed_data
   end
 
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return @store[@front]
   end
 
   def size
-    raise NotImplementedError, "Not yet implemented"
+    @size
   end
 
   def empty?
-    raise NotImplementedError, "Not yet implemented"
+    return @front == -1 || @front == @back
```

Comment:
```suggestion
    return @front == -1
```

If you add 1 element with your algorithm, front will equal back (and you still have 1 thing in the queue).